src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3469,7 +3469,7 @@ private bool HasTopLevelNullabilityConversion(TypeSymbolWithAnnotations source,
             AssignmentKind assignmentKind,
             ParameterSymbol target = null,
             bool reportTopLevelWarnings = true,
-            bool reportNestedWarnings = true,
+            bool reportRemainingWarnings = true,
```

Comment:
>reportRemainingWarnings [](start = 17, length = 23)

Why rename? Does this no longer represent nested warnings?

Reply:
If you convert an `A?` to a `B`, but there is a user-defined conversion from `A` to `B`, the warning produced on the user-defined conversion is in the "remaining" bucket.

The idea is that callers of this API may want to report top-level warnings themselves, so they should set `reportTopLevelWarnings=false` (maybe that should be renamed to `reportOuterTopLevelWarnings`?). For all other warnings, use `reportRemainingWarnings`.

---
In reply to: [245168716](https://github.com/dotnet/roslyn/pull/32090#discussion_r245168716) [](ancestors = 245168716)